File: backend/app/schemas/experience.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional, List

class ExperienceCreate(BaseModel):
    type: str
    title: str
    organization: str
    location: Optional[str] = None
    description: str
    highlights: List[str] = []
    tech_stack: List[str] = []
    start_date: str
    end_date: Optional[str] = None
    is_current: bool = False
    order: int = 0

class ExperienceUpdate(BaseModel):
    title: Optional[str] = None
    organization: Optional[str] = None
    location: Optional[str] = None
    description: Optional[str] = None
    highlights: Optional[List[str]] = None
    tech_stack: Optional[List[str]] = None
    start_date: Optional[str] = None
    end_date: Optional[str] = None
    is_current: Optional[bool] = None
    order: Optional[int] = None





class ExperienceResponse(BaseModel):
    id: str
    type: str
    title: str
    organization: str
    location: Optional[str]
    description: str
    highlights: List[str]
    tech_stack: List[str]
    start_date: str
    end_date: Optional[str]
    is_current: bool
    order: int
    created_at: datetime

File: frontend/src/terminal/commands/experience.js
import { getExperience } from "../../api/experience"
import { formatDate } from "../../utils/formatters"

export default async function experience({ parsed, addOutput, theme, hideTitle = false }) {
  try {
    const type = parsed.flags.work
      ? "work"
      : parsed.flags.education
      ? "education"
      : null

    const data = await getExperience(type)
    if (!hideTitle) {
    addOutput({
      type: "accent",
      text: "── Character Development ─────────────────"
    })

    }
    addOutput({ type: "empty" })

    data.forEach((e, i) => {
      const isLast  = i === data.length - 1
      const end     = e.is_current ? "rn" : formatDate(e.end_date)
      const period  = `${formatDate(e.start_date)} → ${end}`

      const typeTag =
        e.type === "work" ? "grind" : "academia"

      addOutput({
        type: "text",
        html: `<span style="color:${theme.branch}"> [${typeTag}]</span><span style="color:${theme.user}"> ${e.title}</span>`,
      })

      addOutput({
        type: "text",
        html: `<span style="color:${theme.muted}"> ${e.organization}${e.location ? " · " + e.location : ""} </span>`,
      })

      addOutput({
        type: "text",
        html: `<span style="color:${theme.path}"> ${period} </span>`,
      })

      if (e.highlights?.length) {
        e.highlights.forEach(h => {
          addOutput({type: "muted", text: `   • ${h}`
          })
        })
      }

      if (e.tech_stack?.length) {
        addOutput({
          type: "text",
          html: `<span style="color:${theme.muted}"> cooked with → </span> <span style="color:${theme.path}"> ${e.tech_stack.join(", ")}</span>`,
        })
      }

      if (!isLast) {
        addOutput({
          type: "muted",
          text: "   │"
        })
      }

      addOutput({ type: "empty" })
    })

  } catch {
    addOutput({
      type: "error",
      text: "timeline corrupted twin [x_x]"
    })
  }
}